### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by RuslanIbIsh**

### CODE REVIEW

**Review by Ilya-brinchuk — APPROVED** [review]

File: src/main/java/com/iri/spring/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
```
Comment (Sofasmile) [inline src/main/java/com/iri/spring/controller/UserController.java:16]: let's use annotation @ RestController here and then we can remove ResponseBody from methods

File: src/main/java/com/iri/spring/dao/UserDao.java [hunk 2]
```diff
@@ -7,4 +7,6 @@
     void add(User user);
 
     List<User> listUsers();
+
+    User get(Long id);
```
Comment (Sofasmile) [inline src/main/java/com/iri/spring/dao/UserDao.java:11]:
```suggestion
    Optional<User> get(Long id);
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/iri/spring/controller/UserController.java (hunk 1 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/spring/controller/UserController.java]: Yeap! @RestController is the @Controller + @ResponseBody

File: src/main/java/com/iri/spring/dao/UserDao.java (hunk 2 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/spring/dao/UserDao.java]: done

File: src/main/java/com/iri/spring/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -28,6 +29,10 @@ public void add(User user) {
 
     @Override
     public User get(Long id) {
-        return userDao.get(id);
+        Optional<User> optionalUser = userDao.get(id);
+        if (optionalUser.isPresent()) {
+            return optionalUser.get();
+        }
+        throw new RuntimeException("Empty optional user");
```
Comment (Sofasmile) [inline src/main/java/com/iri/spring/service/impl/UserServiceImpl.java:36]:
```suggestion
        userDao.get(id).orElseThrow(() -> new RuntimeException("Empty optional user"));
```

**Review by Sofasmile — APPROVED** [review]

File: src/main/java/com/iri/spring/service/impl/UserServiceImpl.java (hunk 3 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/spring/service/impl/UserServiceImpl.java]: rewrote

**Pull request merged**